File: src/utils/path.ts
import path from "node:path";
import url from "node:url";

export function toPosix(input: string) {
    return path.sep === path.win32.sep
        ? input.replaceAll(path.win32.sep, path.posix.sep)
        : input
}

export function toAbsolute(input: string, cwd: string) {
    const absolutePath = path.isAbsolute(input)
        ? input
        : path.join(cwd, input)

    return toPosix(absolutePath)
}

export function toModuleName(input: string) {
    return path.sep === path.win32.sep
        ? url.pathToFileURL(path.normalize(input)).toString()
        : input
}

File: gradle-plugin/karakum/src/jsMain/kotlin/io/github/sgrishchenko/karakum/util/path.kt
package io.github.sgrishchenko.karakum.util

import node.path.path
import node.url.pathToFileURL

fun toPosix(input: String): String {
    return if (path.sep == path.win32.sep) {
        input.replace(path.win32.sep, path.posix.sep)
    } else {
        input
    }
}

fun toAbsolute(input: String, cwd: String): String {
    val absolutePath = if (path.isAbsolute(input)) {
        input
    } else {
        path.join(cwd, input)
    }

    return toPosix(absolutePath)
}

fun toModuleName(input: String): String {
    return if (path.sep == path.win32.sep) {
        pathToFileURL(path.normalize(input)).toString()
    } else {
        input
    }
}
